### PR TITLE
Made list fields work with classes

### PR DESCRIPTION
**Pull request by alexmorken**
Made list sub-fields work with classes (previously only worked with dicts)

### CODE REVIEW

**Comment by joshfriend** [comment]
So this would fix #291?

File: tests/test_fields.py [hunk 1]
```diff
@@ -412,6 +412,23 @@ def __init__(self, list):
         field = fields.List(fields.String, attribute='foo')
         self.assertEquals(['a', 'b', 'c'], field.output('list', obj))
 
+    def test_list_with_scoped_attribute_on_dict_or_obj(self):
+        class TestObject(object):
+            def __init__(self, list):
+                self.bar = list
```
Comment (joshfriend) [inline tests/test_fields.py]: Please avoid shadowing builtins if possible. You could call this `list_` instead.

Comment (alexmorken) [inline tests/test_fields.py]: Absolutely - I was just following the pattern I thought I saw in the tests...perhaps I misread them a bit.

**Comment by alexmorken** [comment]
Yes - it appears this is a fix for #291

**Comment by joshfriend** [comment]
Well, that's neat! Thanks! :+1:

**Pull request merged**